packages/mdc-list/foundation.js

```diff
@@ -505,6 +473,43 @@ class MDCListFoundation extends MDCFoundation {
     const listSize = this.adapter_.getListItemCount();
     return index >= 0 && index < listSize;
   }
+
+  /**
+   * @param {number} index
+   * @private
+   */
+  setSelectedIndexOnAction_(index) {
+    if (this.isCheckboxList_) {
+      this.toggleCheckboxAtIndex_(index);
+    } else {
+      this.setSelectedIndex(index);
+    }
+  }
+
+  /**
+   * @param {number} index
+   * @private
+   */
+  toggleCheckboxAtIndex_(index) {
+    let isChecked = this.adapter_.isCheckboxCheckedAtIndex(index);
+
+    if (this.toggleCheckbox_) {
```

Comment:
It seems like at this point we should be able to pass this through `setSelectedIndexAction_` to here, since the invocation chain between `handleClick` and this point is private. That should be easier to follow than flipping an instance flag from 2 levels up the call stack.